app/code/community/Payone/Core/Block/Paydirekt/Express/Shortcut.php

```diff
@@ -0,0 +1,249 @@
+ * @link            http://www.fatchip.de
+ */
+class Payone_Core_Block_Paydirekt_Express_Shortcut extends Mage_Core_Block_Template
+{
+    const INIT_CHECKOUT_URL = 'payone_core/paydirektExpress/initCheckout';
+
+    const DEFAULT_LINK_IMAGE = 'https://www.paydirekt.de/presse/medien/logos/ohne-schutzzone/4c_ohne_schutzzone/paydirekt_logo_claim_4c.png';
+
+    /**
+     * Position of "OR" label against shortcut
+     */
+    const POSITION_BEFORE = 'before';
+    const POSITION_AFTER = 'after';
+
+    /** @var Payone_Core_Model_Factory */
+    protected $factory = null;
+
+    /**
+     * Whether the block should be eventually rendered
+     *
+     * @var bool
+     */
+    protected $shouldRender = true;
+
+    /**
+     * Payment method code
+     *
+     * @var string
+     */
+    protected $paymentMethodCode = Payone_Core_Model_System_Config_PaymentMethodCode::WALLETPAYDIREKTEXPRESS;
+
+    /**
+     * @var string
+     */
+    protected $quoteId;
+
+    /**
+     * @var string
+     */
+    protected $customerId;
+
+    /**
+     * @return bool
+     */
+    public function isPaydirektExpressActive()
+    {
+        $config = $this->getConfiguration();
+
+        return !empty($config);
+    }
+
+    /**
+     * @return string
+     */
+    public function getInitUrl()
+    {
+        return Mage::getBaseUrl() . self::INIT_CHECKOUT_URL;
+    }
+
+    /**
+     * @return string
+     */
+    public function getQuoteId()
+    {
+        if ($this->quoteId == null) {
+            $this->quoteId = Mage::getSingleton('checkout/session')->getQuoteId();
+        }
+
+        return $this->quoteId;
+    }
+
+    /**
+     * @param string $quoteId
+     */
+    public function setQuoteId($quoteId)
+    {
+        $this->quoteId = $quoteId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCustomerId()
+    {
+        if ($this->customerId == null) {
+            $this->customerId = Mage::getSingleton('customer/session')->getCustomerId();
+        }
+
+        return $this->customerId;
+    }
+
+    /**
+     * @param string $customerId
+     */
+    public function setCustomerId($customerId)
+    {
+        $this->customerId = $customerId;
+    }
+
+    /**
+     * @return Mage_Core_Block_Abstract
+     */
+    protected function _beforeToHtml()
+    {
+        $result = parent::_beforeToHtml();
+        $quote = Mage::getSingleton('checkout/session')->getQuote();
+
+        // check payment method availability
+        $methodInstance = Mage::helper('payment')->getMethodInstance($this->paymentMethodCode);
+        if (!$methodInstance || !$methodInstance->isAvailable($quote)) {
+            $this->shouldRender = false;
+            return $result;
+        }
+
+        // check visibility button on shopping cart
+        $shortcutOnShoppingCart = $this->getHelperConfig()->getConfigGeneral($quote->getStoreId())->getPaymentPaydirektExpressCheckout()->getPaydirektExpressCheckoutVisibleOnCart();
```

Comment:
Just to say. Codes like this will work indeed, but are hardly readable IMO. I would prefer to fetch needed values/variables before so the final call will match into 80 colums. If you use it this way, at least it should be made more readable like this:

`
$shortcutOnShoppingCart = 
  $this->getHelperConfig()
 ->getConfigGeneral($quote->getStoreId())
 ->getPaymentPaydirektExpressCheckout()
 ->getPaydirektExpressCheckoutVisibleOnCart();
`